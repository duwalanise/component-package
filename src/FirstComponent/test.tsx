import { render } from "@testing-library/react";
import { FirstComponent } from ".";

describe("<FirstComponent />", () => {
  it("should render default title and body", () => {
    const { getByRole, getByText, container } = render(<FirstComponent />);

    getByRole("heading", { name: /hello, world!/i });
    getByText(/i am the first component of this lib/i);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render title and body passed as props", () => {
    const { getByRole, getByText } = render(
      <FirstComponent title='Custom Title' body='Custom Body' />,
    );

    getByRole("heading", { name: /custom title/i });
    getByText(/custom body/i);
  });
});
