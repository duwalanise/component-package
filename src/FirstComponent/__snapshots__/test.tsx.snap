// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirstComponent /> should render default title and body 1`] = `
<div>
  <h1>
    Hello, World!
  </h1>
  <p>
    I am the first component of this lib
  </p>
</div>
`;
